src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public side1: number,
+    public side2: number,
+    public side3: number,
+  ) {
+    if (this.side1 <= 0 || this.side2 <= 0 || this.side3 <= 0) {
```

Comment:
You don't need to write `this` inside of the constructor, it's redundant. The same in other constructors.
```suggestion
    if (side1 <= 0 || side2 <= 0 || side3 <= 0) {
```

Reply:
I did before without public